services:
  db:
      image: postgres
      environment:
        - DB_NAME=${DB_NAME}
        - DB_USER=${DB_USER}
        - DB_PASSWORD=${DB_PASSWORD}
      volumes:
        - ./data/db:/var/lib/postgresql/data
      ports:
        - "15432:5432"
      healthcheck:
        test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME} -p 5432"]
        interval: 5s
        timeout: 5s
        retries: 5
      networks:
        - djangonetwork
  pgadmin:
      image: dpage/pgadmin4
      environment:
        - PGADMIN_DEFAULT_EMAIL=${PGADMIN_DEFAULT_EMAIL}
        - PGADMIN_DEFAULT_PASSWORD=${PGADMIN_DEFAULT_PASSWORD}
      ports:
        - "16543:80"
      depends_on:
        db:
          condition: service_healthy
      networks:
        - djangonetwork
  app:
    build:
      context: .
      args:
        - DEV=true
    ports:
      - "8000:8000"
    volumes:
      - ./app:/app
    environment:
      - SECRET_KEY=${SECRET_KEY}
      - DB_NAME=${DB_NAME}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
    depends_on:
      db:
        condition: service_healthy
    networks:
      - djangonetwork
    command: >
      sh -c "python manage.py runserver 0.0.0.0:8000"
networks: 
  djangonetwork:
    driver: bridge